nimble/host/mesh/src/prov.c

```diff
@@ -249,8 +257,10 @@ static void reset_link(void)
 #endif
 	link.rx.prev_id = XACT_NVAL;
 
-	if (bt_pub_key_get()) {
-		atomic_set_bit(link.flags, LOCAL_PUB_KEY);
+	err = bt_pub_key_gen(&pub_key_cb);
```

Comment:
At least in Zephyr `bt_pub_key_gen()` stores the callback persistently in a linked list. As long as we call this from `init()` there's no problem (it's only called once) but when you call it from `reset()` you'll end up with a corrupted linked list. I'm currently looking at how/if we should change our `bt_pub_key_gen()` implementation to clear the list when generation is complete (i.e. the callbacks wouldn't be persistently stored)